### PR TITLE
Add integration test cases - Closes #887

### LINKED ISSUE #887: Review integration tests - Account history export
### Description

Review integration tests and add the missing test cases for account history export.

Make sure the endpoint `/api/v2/transactions/export` are properly tested.

### Acceptance Criteria

- All the endpoints have integration tests written
- All the scenarios are covered

### PR DESCRIPTION
**Pull request by nagdahimanshu**
### What was the problem?
This PR resolves #887 

### How was it solved?
- [x] Add integration tests for following endpoint:
  - [x] `/api/v2/transactions/export` (http + rpc)
  - [x] `/api/v2/exports/:filename` (http)
- [x] Add `http-exports` in namespaces.js to enable `get.transactioins,export`
- [x] Fix swagger for endpoint `api/v2/exports/{filename}`

### How was it tested?
Local + Jenkins

### CODE REVIEW

File: tests/integration/api_v2/http/export.test.js [hunk 1]
```diff
@@ -16,26 +16,122 @@
...
 	});
 
-	it('File not exists -> 404 NOT_FOUND', async () => {
-		const invalidFile = 'invalid.csv';
-		const response = await api.get(`${endpoint}/${invalidFile}`, 404);
-		expect(response).toMap(notFoundSchema);
+	describe('File is ready to export', () => {
+		it('scheduled from account address -> return 200 OK', async () => {
+			// Add delay of 10 seconds
+			await new Promise(resolve => setTimeout(resolve, 10000));
```
Comment (MichalTuleja) [inline tests/integration/api_v2/http/export.test.js:72]: Did you consider adding a setInterval and wait for the 200 OK response in 500ms intervals?
A helper function that resolves only if the result is correct. Otherwise it hits the timeout of the tests.

File: tests/integration/api_v2/http/export.test.js [hunk 2]
```diff
@@ -16,26 +16,122 @@
...
+			expect(response).toMap(goodRequestSchema);
+			expect(response.data).toBeInstanceOf(Object);
+			expect(response.data).toMap(exportSchema);
+			expect(response.meta).toMap(metaSchema);
+			expect(response.meta).toEqual(expect.objectContaining(expected));
+		});
+
+		it('scheduled from account from publicKey -> return 200 OK', async () => {
+			// Add delay of 10 seconds
+			await new Promise(resolve => setTimeout(resolve, 10000));
```
Comment (MichalTuleja) [inline tests/integration/api_v2/http/export.test.js:85]: Same as above.

File: tests/integration/api_v2/rpc/export.test.js [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+			expect(result.data).toMap(exportSchemaAccepted);
+			expect(result.meta).toMap(metaSchema);
+			expect(result.meta).toEqual(expect.objectContaining(expected));
+		});
+	});
+
+	describe('File is ready to export', () => {
+		it('scheduled from account address -> return 200 OK', async () => {
+			// Add delay of 10 seconds
+			await new Promise(resolve => setTimeout(resolve, 10000));
```
Comment (MichalTuleja) [inline tests/integration/api_v2/rpc/export.test.js:82]: Same as above.

File: tests/integration/api_v2/rpc/export.test.js [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+			const { result } = response;
+			expect(result).toMap(goodRequestSchema);
+			expect(result.data).toMap(exportSchema);
+			expect(result.meta).toMap(metaSchema);
+			expect(result.meta).toEqual(expect.objectContaining(expected));
+		});
+
+		it('scheduled from account publicKey -> return 200 OK', async () => {
+			// Add delay of 10 seconds
+			await new Promise(resolve => setTimeout(resolve, 10000));
```
Comment (MichalTuleja) [inline tests/integration/api_v2/rpc/export.test.js:99]: Same as above.

File: tests/integration/api_v2/http/export.test.js [hunk 5]
```diff
@@ -16,26 +16,122 @@
...
+			expect(response.data).toMap(exportSchema);
+			expect(response.meta).toMap(metaSchema);
+			expect(response.meta).toEqual(expect.objectContaining(expected));
+		});
+	});
+
+	describe('Download csv file -> returns 200 OK', () => {
+		it('scheduled from account address -> 200 OK', async () => {
+			const validFileName = `transactions_${refTransaction1.sender.address}_${startDate}_${endDate}.csv`;
+			await api.get(`${baseUrlV2}/exports/${validFileName}`, 200);
```
Comment (MichalTuleja) [inline tests/integration/api_v2/http/export.test.js:100]: Can you also add the CSV content checks?

File: tests/integration/api_v2/http/export.test.js [hunk 6]
```diff
@@ -16,26 +16,122 @@
...
+		});
+
+		it('scheduled from account publicKey -> 200 OK', async () => {
+			const validFileName = `transactions_${refTransaction2.sender.address}_${startDate}_${endDate}.csv`;
+			await api.get(`${baseUrlV2}/exports/${validFileName}`, 200);
```
Comment (MichalTuleja) [inline tests/integration/api_v2/http/export.test.js:105]: Same as above.

**Review by MichalTuleja — CHANGES_REQUESTED** [review]

File: tests/schemas/api_v2/export.schema.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+	interval: Joi.string().pattern(/^\b((\d{4})-((1[012])|(0?[1-9]))-(([012][1-9])|([123]0)|31))(:((\d{4})-((1[012])|(0?[1-9]))-(([012][1-9])|([123]0)|31)))?\b$/).required(),
+};
+
+const metaSchema = {
+	ready: Joi.boolean().required(),
+};
+
+const goodRequestSchema = {
+	data: Joi.object().required(),
+	meta: Joi.object().required(),
```
Comment (sameersubudhi) [inline tests/schemas/api_v2/export.schema.js:38]:
```suggestion
	meta: Joi.object(metaSchema).required(),
```

File: tests/schemas/api_v2/export.schema.js [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+	interval: Joi.string().pattern(/^\b((\d{4})-((1[012])|(0?[1-9]))-(([012][1-9])|([123]0)|31))(:((\d{4})-((1[012])|(0?[1-9]))-(([012][1-9])|([123]0)|31)))?\b$/).required(),
+	fileName: Joi.string().required(),
+	fileUrl: Joi.string().required(),
+};
+
+const exportSchemaAccepted = {
+	address: Joi.string().pattern(/^lsk[a-hjkm-z2-9]{38}$/).required(),
+	publicKey: Joi.string().pattern(/^([A-Fa-f0-9]{2}){32}$/).optional(),
+	interval: Joi.string().pattern(/^\b((\d{4})-((1[012])|(0?[1-9]))-(([012][1-9])|([123]0)|31))(:((\d{4})-((1[012])|(0?[1-9]))-(([012][1-9])|([123]0)|31)))?\b$/).required(),
+};
```
Comment (sameersubudhi) [inline tests/schemas/api_v2/export.schema.js:30]:
```suggestion
const regex = {
	address: /^lsk[a-hjkm-z2-9]{38}$/,
	publicKey: /^([A-Fa-f0-9]{2}){32}$/,
	interval: /^\b((\d{4})-((1[012])|(0?[1-9]))-(([012][1-9])|([123]0)|31))(:((\d{4})-((1[012])|(0?[1-9]))-(([012][1-9])|([123]0)|31)))?\b$/,
	fileName: /^\btransactions_(lsk[a-hjkm-z2-9]{38})_((\d{4})-((1[012])|(0?[1-9]))-(([012][1-9])|([123]0)|31))_((\d{4})-((1[012])|(0?[1-9]))-(([012][1-9])|([123]0)|31))\.csv\b$/g,
	fileUrl: /^\/api\/v2\/exports\/transactions_(lsk[a-hjkm-z2-9]{38})_((\d{4})-((1[012])|(0?[1-9]))-(([012][1-9])|([123]0)|31))_((\d{4})-((1[012])|(0?[1-9]))-(([012][1-9])|([123]0)|31))\.csv$/g,
};

const exportSchemaAccepted = {
	address: Joi.string().pattern(regex.address).required(),
	publicKey: Joi.string().pattern(regex.publicKey).optional(),
	interval: Joi.string().pattern(regex.interval).required(),
};

const exportSchema = {
	...exportSchemaAccepted,
	fileName: Joi.string().pattern(regex.fileName).required(),
	fileUrl: Joi.string().pattern(regex.fileUrl).required(),
};
```

File: tests/integration/api_v2/http/export.test.js [hunk 9]
```diff
@@ -16,26 +16,122 @@
...
 } = require('../../../schemas/httpGenerics.schema');
 
 const baseUrl = config.SERVICE_ENDPOINT;
 const baseUrlV2 = `${baseUrl}/api/v2`;
-const endpoint = `${baseUrlV2}/exports`;
 
-const expectedResponse = '"Date";"Time";"Amount LSK";"Fee LSK";"Module:Asset";"Module:Asset Name";"Sender";"Recipient";"Sender Public Key";"Recipient Public Key";"Block Height";"Note";"Transaction ID"\n"2021-08-20";"10:53:06";"10.00000000";"0.00000000";"2:0";"token:transfer";"lskdxc4ta5j43jp9ro3f8zqbxta9fn6jwzjucw7yt";"lskqz6gpqfu9tb5yc2jtqmqvqp3x8ze35g99u2zfd";"0fe9a3f1a21b5530f27f87a414b549e79a940bf24fdf2b2f05e7f22aeeecc86a";"10bdf57ee21ff657ab617395acab81814c3983f608bf6f0be6e626298225331d";413;;"28fcc24bdbdf25e4ae6350fdac6af5c08f5a13bece10a34f27478f07569ecef0"\n"2021-08-20";"10:54:56";"165.00000000";"0.00119000";"1000:0";"legacyAccount:reclaimLSK";"lskqz6gpqfu9tb5yc2jtqmqvqp3x8ze35g99u2zfd";;"10bdf57ee21ff657ab617395acab81814c3983f608bf6f0be6e626298225331d";;424;;"6cff643daaa2bd1112d1b4591abef3e62f9e4f6e37a260fcd7508ce6a06f061c"\n';
+describe('Export API', () => {
+	const startDate = '2021-01-10';
+	const endDate = '2021-11-30';
```
Comment (sameersubudhi) [inline tests/integration/api_v2/http/export.test.js:36]: Kindly consider the following change:
```
const moment = require('moment');

const config = require('../../../config');
const exportConfig = require('../../../../services/export/config');

const {
	api,
} = require('../../../helpers/api');

const {
	metaSchema,
	exportSchema,
	exportSchemaAccepted,
	goodRequestSchema,
} = require('../../../schemas/api_v2/export.schema');

const {
	notFoundSchema,
	badRequestSchema,
} = require('../../../schemas/httpGenerics.schema');

const baseUrl = config.SERVICE_ENDPOINT;
const baseUrlV2 = `${baseUrl}/api/v2`;

describe('Export API', () => {
	const startDate = moment('2021-01-10').format(exportConfig.csv.dateFormat);
	const endDate = moment('2021-11-30').format(exportConfig.csv.dateFormat);
```

File: tests/integration/api_v2/rpc/export.test.js [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+	goodRequestSchema,
+} = require('../../../schemas/api_v2/export.schema');
+
+const wsRpcUrl = `${config.SERVICE_ENDPOINT}/rpc-v2`;
+const requestTransactionExport = async (params) => request(wsRpcUrl, 'get.transactions.export', params);
+const requestTransactions = async (params) => request(wsRpcUrl, 'get.transactions', params);
+
+describe('Export API', () => {
+	const startDate = '2021-01-10';
+	const endDate = '2021-11-30';
```
Comment (sameersubudhi) [inline tests/integration/api_v2/rpc/export.test.js:38]: As above.
```suggestion
const moment = require('moment');

const config = require('../../../config');
const exportConfig = require('../../../../services/export/config');

const {
	request,
} = require('../../../helpers/socketIoRpcRequest');

const {
	jsonRpcEnvelopeSchema,
	invalidParamsSchema,
	invalidRequestSchema,
} = require('../../../schemas/rpcGenerics.schema');

const {
	metaSchema,
	exportSchema,
	exportSchemaAccepted,
	goodRequestSchema,
} = require('../../../schemas/api_v2/export.schema');

const wsRpcUrl = `${config.SERVICE_ENDPOINT}/rpc-v2`;
const requestTransactionExport = async (params) => request(wsRpcUrl, 'get.transactions.export', params);
const requestTransactions = async (params) => request(wsRpcUrl, 'get.transactions', params);

describe('Export API', () => {
	const startDate = moment('2021-01-10').format(exportConfig.csv.dateFormat);
	const endDate = moment('2021-11-30').format(exportConfig.csv.dateFormat);
```

**Review by sameersubudhi — CHANGES_REQUESTED** [review]
Kindly consider the following changes as well.

**Comment by sameersubudhi** [comment]
@nagdahimanshu Also, please ensure to add a few more test cases to cover the following conditions, both positive & negative scenarios:
```
	validParamPairings: [
		['address'],
		['publicKey'],
		['address', 'interval'],
		['publicKey', 'interval'],
	]
```

File: tests/integration/api_v2/http/export.test.js (hunk 5 above)
Comment (nagdahimanshu) [reply tests/integration/api_v2/http/export.test.js]: I have added a check to validate the CSV content.

File: tests/integration/api_v2/rpc/export.test.js [hunk 11]
```diff
@@ -0,0 +1,173 @@
...
+	beforeAll(async () => {
+		const response1 = await requestTransactions({ limit: 1 });
+		[refTransaction1] = response1.result.data;
+
+		const response2 = await requestTransactions({ limit: 1, offset: 1 });
+		[refTransaction2] = response2.result.data;
+	});
+
+	describe('Schedule file export', () => {
+		it('Schedule file export from account address -> return 202 Accepted', async () => {
```
Comment (sameersubudhi) [inline tests/integration/api_v2/rpc/export.test.js:54]: Please add two more test cases:
1. Only `address` in request params
2. Only `publicKey` in request params 

Also, please update the test case description. `202 ACCEPTED`, `200 OK` are all HTTP codes.

File: tests/integration/api_v2/http/export.test.js [hunk 12]
```diff
@@ -13,29 +13,190 @@
...
+	beforeAll(async () => {
+		const response1 = await api.get(`${baseUrlV2}/transactions?limit=1`);
+		[refTransaction1] = response1.data;
+
+		const response2 = await api.get(`${baseUrlV2}/transactions?limit=1&offset=1`);
+		[refTransaction2] = response2.data;
+	});
+
+	describe('Schedule file export', () => {
+		it('Schedule file export from account address -> return 202 Accepted', async () => {
```
Comment (sameersubudhi) [inline tests/integration/api_v2/http/export.test.js:65]: Please add similar test cases here as well.

**Review by sameersubudhi — CHANGES_REQUESTED** [review]

**Review by MichalTuleja — APPROVED** [review]
LGTM 👍

File: tests/integration/api_v2/http/export.test.js [hunk 13]
```diff
@@ -53,16 +53,24 @@ describe('Export API', () => {
...
 		[refTransaction1] = response1.data;
 
 		const response2 = await api.get(`${baseUrlV2}/transactions?limit=1&offset=1`);
 		[refTransaction2] = response2.data;
+
+		const response3 = await api.get(`${baseUrlV2}/transactions?limit=1&offset=2`);
+		[refTransaction3] = response3.data;
+
+		const response4 = await api.get(`${baseUrlV2}/transactions?limit=1&offset=3`);
+		[refTransaction4] = response4.data;
```
Comment (sameersubudhi) [inline tests/integration/api_v2/http/export.test.js]: This could be reduced to a single API invocation.
```
		const response = await api.get(`${baseUrlV2}/transactions?limit=4`);
		[refTransaction1, refTransaction2, refTransaction3, refTransaction4] = response.data;
```

File: tests/integration/api_v2/rpc/export.test.js [hunk 14]
```diff
@@ -42,16 +42,24 @@ describe('Export API', () => {
...
 		[refTransaction1] = response1.result.data;
 
 		const response2 = await requestTransactions({ limit: 1, offset: 1 });
 		[refTransaction2] = response2.result.data;
+
+		const response3 = await requestTransactions({ limit: 1, offset: 2 });
+		[refTransaction3] = response3.result.data;
+
+		const response4 = await requestTransactions({ limit: 1, offset: 3 });
+		[refTransaction4] = response4.result.data;
```
Comment (sameersubudhi) [inline tests/integration/api_v2/rpc/export.test.js]: Same as above.
```
		const response = await requestTransactions({ limit: 4 });
		[refTransaction1, refTransaction2, refTransaction3, refTransaction4] = response.result.data;
```

**Review by sameersubudhi — APPROVED** [review]
LGTM overall 👍
With some optional suggestions.
